easybuild/easyblocks/s/star_ccm.py

```diff
@@ -0,0 +1,114 @@
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing STAR-CCM+, implemented as an easyblock
+"""
+import glob
+import os
+import tempfile
+
+import easybuild.tools.environment as env
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import change_dir
+from easybuild.tools.run import run_cmd
+
+
+class EB_STAR_minus_CCM_plus_(EasyBlock):
+    """Support for building/installing STAR-CCM+."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialise STAR-CCM+ easyblock."""
+        super(EB_STAR_minus_CCM_plus_, self).__init__(*args, **kwargs)
+        self.starccm_subdir = None
+        self.starview_subdir = None
+
+    def configure_step(self):
+        """No configuration procedure for STAR-CCM+."""
+        pass
+
+    def build_step(self):
+        """No build procedure for STAR-CCM+."""
+        pass
+
+    def install_step(self):
+        """Custom install procedure for STAR-CCM+."""
+        cands = glob.glob("./STAR-CCM+%s_*.sh" % self.version)
+        if len(cands) == 1:
+            install_script = cands[0]
+
+            env.setvar('CHECK_DISK_SPACE', 'OFF')
+            env.setvar('IATEMPDIR', tempfile.mkdtemp())
+
+            cmd = ' '.join([
+                install_script,
+                "-i silent",
+                "-DINSTALLDIR=%s" % self.installdir,
+                "-DINSTALLFLEX=false",
+                "-DADDSYSTEMPATH=false",
+            ])
+            run_cmd(cmd, log_all=True, simple=True)
+        else:
+            raise EasyBuildError("Found no or multiple possible matches for STAR-CCM+ installation script: %s", cands)
+
+    def find_starccm_subdirs(self):
+        """Determine subdirectory of install directory in which STAR-CCM+ was installed."""
+        cwd = change_dir(self.installdir)
+        cands = glob.glob(os.path.join(self.version + '*', 'STAR-CCM+%s*' % self.version))
+        if len(cands) == 1:
+            self.starccm_subdir = cands[0]
+            self.log.info("Found STAR-CCM+ subdirectory: %s", self.starccm_subdir)
+            self.starview_subdir = os.path.join(os.path.dirname(self.starccm_subdir), 'STAR-View+%s' % self.version)
+        else:
+            raise EasyBuildError("Failed to determine the STAR-CCM+ subdirectory in %s: %s", self.installdir, cands)
```

Comment:
dry-run case

Reply:
maybe this flow should be put in the framework? I've seen it in multiple easyblock ?

Reply:
It actually already exists: [`find_glob_pattern` from `easybuild.tools.filetools`](https://docs.easybuild.io/en/latest/api/easybuild.tools.filetools.html#easybuild.tools.filetools.find_glob_pattern)

Reply:
fixed in b2dc80685